packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,308 @@
+      horizontal: {
+        size: {
+          S: size(72),
+          M: size(106)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(168),
+          S: size(184),
+          M: size(200),
+          L: size(216),
+          XL: size(232)
+        }
+      }
+    }
+  },
+  minHeight: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(72),
+          M: size(106)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(100),
+          S: size(128),
+          M: size(144),
+          L: size(160),
+          XL: size(192)
+        }
+      }
+    }
+  },
+  maxWidth: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(168),
+          S: size(184),
+          M: size(200),
+          L: size(216),
+          XL: size(232)
+        }
+      }
+    }
+  },
+  minWidth: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(100),
+          S: size(128),
+          M: size(144),
+          L: size(160),
+          XL: size(192)
+        }
+      }
+    }
+  },
+  outlineColor: {
+    default: 'focus-ring',
+    forcedColors: 'Highlight'
+  },
+  padding: {
+    orientation: {
+      horizontal: {
+        default: 16
+      }
+    }
+  },
+  position: 'relative'
+});
+
+const selectBoxContentStyle = style({
+  alignContent: 'center',
+  alignItems: 'center',
+  display: 'grid',
+  gap: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: 0,
+          M: size(2)
+        },
+        vertical: size(8)
+      }
+    }
+  },
+  gridTemplateAreas: {
+    orientation: {
+      horizontal: ['icon label', 'icon description'],
+      vertical: ['. icon .', '. label .']
+    }
+  },
+  gridTemplateColumns: {
+    orientation: {
+      horizontal: '48px 1fr'
+    }
+  },
+  height: 'full',
+  justifyItems: {
+    orientation: {
+      horizontal: 'flex-start',
+      vertical: 'center'
+    }
+  },
+  paddingStart: {
+    orientation: {
+      horizontal: size(12),
+      vertical: 0
+    }
+  },
+  width: 'auto'
+});
+
+const selectBoxIconStyle = style({
+  color: {
+    default: 'red-600',
+    isDisabled: 'disabled'
+  },
+  gridArea: 'icon',
+  marginBottom: {
+    orientation: {
+      horizontal: 0,
+      vertical: 8
+    }
+  },
+  flexShrink: 0
+});
+
+const selectBoxLabelStyle = style({
+  gridArea: 'label',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  },
+  fontSize: {
+    size: {
+      XS: 'body-sm',
+      M: 'body-lg',
+      L: 'body-xl',
+      XL: 'body-2xl'
+    }
+  }
+});
+
+const selectorStyle = style({
+  display: 'block',
+  position: 'absolute',
+  top: {
+    orientation: {
+      vertical: 16,
+      horizontal: '50%'
+    }
+  },
+  right: 16,
```

Comment:
just a heads up, i think this is getting moved to top left following how cards looks
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/s2/src/Card.tsx#L202
may also be worthwhile to compare styles from cards to this, I suspect there's a fair amount of overlap